.github/workflows/build-push-multi-arch-docker-image.yml

```
@@ -0,0 +1,139 @@
+name: Multi-arch Docker Image Build & Push
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: Tag to be assigned when pushing the image to ECR repo
+        required: true
+        type: string
+      latest:
+        description: Tag as latest, if enabled image will be also tagged as 'latest' in ECR repo
+        required: false
+        type: boolean
+        default: false
+      image-target:
+        description: Target for image build, selects stage for a multi-stage Dockefile
+        required: true
+        type: string
+      Dockerfile:
+        description: Dockerfile to be used for the build, accepts path to file
+        required: true
+        type: string
+      build-context:
+        description: Build context to be used by 'docker build' command
+        required: false
+        type: string
+        default: .
+    secrets:
+      AWS_ECR_REPOSITORY_URL:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      AWS_REGION:
+        required: true
+
+jobs:
+  build-push:
+    name: Docker image to ECR
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - amd64
+          - arm64
```

Can this be an input?

The user can provider the input as such:

`platforms: '["amd64", "arm64"]'`

Then you can load it in as so:
```suggestion
      matrix:
        platform: ${{ fromJSON(inputs.platforms) }}
```

This of course, means adding a new input `platforms`

---

I thought about this possible solution but the introduction of the platforms as an input was overcomplicating the manifest creation in the merge job then.

---

UPDATE:

Had an offline conversation with Gabrielle. We decided to rename this action to `build-push-amd64-arm64-docker-image` or something similar.

Though there are many available platforms, updating this action to be platform agnostic was making it quite and complicated, and more importantly, not worth the effort since the other architectures we will not be using as much (or at all) at Xanadu.

So the action is staying as is, with the name being scoped down. :+1: